### PR TITLE
[RFR] Fix firefox issue with headers.forEach and make type check optional

### LINKED ISSUE #58: errorCode global interceptor
Hi there,
thanks for the restful.js lib, I'm finding it very useful.

I have a question re. making a global interceptor for errors of different types.

I want to handle rest errors globally, consistently, but can not figure out how to using restful.js.

If I were using jquery's ajax methods, I would simply use:

	//set a global method to handle 500 errors with ajax calls

	$( document ).ajaxError(function(err) {
		console.log("error: ", err);
          bootbox.dialog({
			message: "Ajax request failed with a 500 error - ",
			className: "danger",
			buttons: {
				danger: {
				  label: "OK",
				  className: "btn-danger",
				  callback: plenty_admin.HELPER.hideLoadingOverlay
				}
			},
			 onEscape: plenty_admin.HELPER.hideLoadingOverlay,
			 callback: plenty_admin.HELPER.hideLoadingOverlay
		});
	});
	

Can you possibly provide a similar example for use with a pretty basic instanciation of restful.js?

Thanks so much!

Jamie

### LINKED ISSUE #66: Firefox fetch issue
I'm getting this error in FF, when sending requests:
```
TypeError: response.headers.forEach is not a function
Stack trace:
exports.default/</</<@webpack:///./~/restful.js/dist/es5/http/fetch.js?:30:17
```
I'm using webpack, and have fetch and es6-promise plugged in, like described here:
https://gist.github.com/Couto/b29676dd1ab8714a818f

Any ideas what the problem is?

### PR DESCRIPTION
**Pull request by RobinBressan**
- [x] Fixes #66, #60
- [x] Fixes #58

### CODE REVIEW

**Comment by yborunov** [comment]
That should work, great job! :+1:

File: build/restful.fetch.js [hunk 1]
```diff
@@ -1,7 +1,11 @@
+import assign from 'object-assign';
 import restful from '../src';
 import fetchBackend from '../src/http/fetch';
 import 'whatwg-fetch';
 
-export default function(baseUrl, httpBackend = fetchBackend(fetch)) {
+export default assign((baseUrl, httpBackend = fetchBackend(fetch)) => {
```
Comment (jpetitcolas) [inline build/restful.fetch.js]: Not used.

**Comment by jpetitcolas** [comment]
:+1:

**Comment by yborunov** [comment]
Problem is still occurring in FF v43.0.3 because it doesn't have keys method in Headers object :(

**Comment by simonchrz** [comment]
:+1: still valid this issue

**Comment by RobinBressan** [comment]
@yborunov It works with the latest firefox version (44). Before this version, I didn't find a way to retrieve all response headers without the `key` or `forEach` method.

**Comment by simonchrz** [comment]
@RobinBressan i think you forgot to recompile the restful.standalone file ?https://github.com/marmelab/restful.js/blob/master/dist/restful.standalone.js#L7158
--> the es5 files look headers.forEach clean :-)

**Comment by RobinBressan** [comment]
@simonchrz This line is from the fetch polyfill which is included in this build

**Comment by simonchrz** [comment]
k, thx. works with firefox >= v44

**Comment by yborunov** [comment]
@RobinBressan yes, I know this's is a very tricky problem, since all browsers have different implementation of Headers interface, the only possible way I see is to toggle between different methods depend on what is available for now. I created a PR with my solution: https://github.com/marmelab/restful.js/pull/77. Please, take a look when you have time.

**Pull request merged**